### ISSUE TITLE
[bug]: RuntimeError: Attempting to deserialize object on a CUDA device but torch.cuda.is_available() is False

### ISSUE THREAD

**Issue by soten355** [invoke-ai/InvokeAI#2844]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### OS

macOS

### GPU

mps

### VRAM

8GB

### What happened?

Attempting to load InvokeAI for the first time after installation and I get this Runtime error:

> RuntimeError: Attempting to deserialize object on a CUDA device but torch.cuda.is_available() is False. If you are running on a CPU-only machine, please use torch.load with map_location=torch.device('cpu') to map your storages to the CPU.

No idea how to change a setting without re-installing and even then I get the same error.

### Additional context

All of my models are .ckpt or .pt and are being imported from a seperate folder.

**Comment by mindfu23** [comment]
Just received this message on updating to Invoke 2.3.2. I'm on an M2 macbook pro.

**Comment by mindfu23** [comment]
...I have just found a solution that resolves the issue via Chat GPT. : ) This appears to be the result of Mac M2s having no GPU to speak of. The solution comes from editing the model_manager.py file. 

This chunk starts at line 871, the revised line is #875.

/Users/<yourusername>/invokeai/.venv/lib/python3.10/site-packages/ldm/invoke

```
        # another round of heuristics to guess the correct config file.
        checkpoint = (
            safetensors.torch.load_file(model_path)
            if model_path.suffix == ".safetensors"
            else torch.load(model_path)
```

Replace last line there with:

` else torch.load(model_path, map_location=torch.device('cpu'))`

...oh, and press the magic “Command” + “Shift” + “.” (period) keys to see the hidden .venv folder of course. 

Worked for me.

**Comment by jameslanman** [comment]
Half of my legacy ckpts won't load due to this error. I tried @mindfu23 s fix(thanks!) but it didn't fix the issue. Any ideas for the mac users?

**Comment by sr1234** [comment]
Running into this on an M1 Macbook Pro 16.  I had a completely working version of InvokeAI with a lot of models, including some that I created.  Tried both manual install and using the installer.  Both gave this error.  Also, I noticed somewhere along the way that it thinks there is no GPU.  Concerned about that because M1 Macbook Pros do have a GPU. 

Now the previous version I had installed won't load either.  I did try to install it in an alternative directory, just assuming installations would be self-contained.  That was a mistake.  Any help would be really appreciated.
